test/e2e/logcollector.go

```diff
@@ -0,0 +1,231 @@
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubectl/pkg/describe"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	expv1 "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Metal3LogCollector struct{}
+
+// CollectMachineLog collects specific logs from machines.
+func (Metal3LogCollector) CollectMachineLog(ctx context.Context, cli client.Client, m *clusterv1.Machine, outputPath string) error {
```

Comment:
It is more common to use a pointer syntax here, like `func (m *Metal3LogCollector)`.
I'm curious, though, why an empty struct is needed here :-?

Reply:
I created a `logcollector.go` file and moved some logging functions from `common.go` to this new file for more logical grouping. I copied the function as is and I can't say anything about why the function is written this way. But I can change it based on @mquhuy 's suggestion if there are no counter arguments.